aiohttp/connector.py

```diff
@@ -69,6 +72,39 @@
     from .tracing import Trace
 
 
+def _convert_hosts_to_addr_infos(
+    hosts: List[Dict[str, Any]]
+) -> List[aiohappyeyeballs.AddrInfoType]:
+    """Converts the list of hosts to a list of addr_infos.
+
+    The list of hosts is the result of a DNS lookup. The list of
+    addr_infos is the result of a call to `socket.getaddrinfo()`.
+    """
+    addr_infos: List[aiohappyeyeballs.AddrInfoType] = []
+    for hinfo in hosts:
+        host = hinfo["host"]
+        is_ipv6 = helpers.is_ipv6_address(host)
+        family = socket.AF_INET6 if is_ipv6 else socket.AF_INET
+        addr = (host, hinfo["port"], 0, 0) if is_ipv6 else (host, hinfo["port"])
+        addr_infos.append(
+            (family, socket.SOCK_STREAM, socket.IPPROTO_TCP, hinfo["hostname"], addr)
+        )
+    return addr_infos
+
+
+def _convert_local_addr_to_addr_infos(
+    local_addr: Optional[Tuple[str, int]]
+) -> Optional[List[aiohappyeyeballs.AddrInfoType]]:
+    """Convert a local_addr tuple to a list of addr_infos."""
+    if local_addr is None:
+        return None
+    host, port = local_addr
+    is_ipv6 = helpers.is_ipv6_address(host)
+    family = socket.AF_INET6 if is_ipv6 else socket.AF_INET
+    addr = (host, port, 0, 0) if is_ipv6 else (host, port)
+    return [(family, socket.SOCK_STREAM, socket.IPPROTO_TCP, "", addr)]
```

Comment:
This isn't aiohttp specific so it could be moved to aiohappyeyeballs

Reply:
This one is a bit of a special case though since ipv6 is always a 4 arg tuple, and this only takes a 2 arg tuple

Reply:
Need to see how cpython handles this

Reply:
https://github.com/python/cpython/blob/9cdf05bc28c5cd8b000b9541a907028819b3d63e/Lib/asyncio/base_events.py#L1472

Reply:
Moved to the lib since its useful elsewhere and not aiohttp specific